darts/ad/scorers.py

```diff
@@ -0,0 +1,1582 @@
+        else:
+            for (s1, s2) in zip(list_series_1, list_series_2):
+                _sanity_check_2series(s1, s2)
+                s1, s2 = self._check_probabilistic_case(s1, s2)
+                anomaly_scores.append(self._score_core(s1, s2))
+
+        if len(anomaly_scores) == 1 and not isinstance(series_1, Sequence):
+            return anomaly_scores[0]
+        else:
+            return anomaly_scores
+
+    def _check_window_size(self, series: TimeSeries):
+        """Checks if the parameter window is less or equal to the length of the given series"""
+
+        raise_if_not(
+            self.window <= len(series),
+            f"Window size {self.window} is greater than the targeted series length {len(series)}, \
+            must be lower or equal.",
+        )
+
+    @abstractmethod
+    def _fit_core(self, input: Any) -> Any:
+        pass
+
+    def fit(
+        self,
+        series_1: Union[TimeSeries, Sequence[TimeSeries]],
+        series_2: Union[TimeSeries, Sequence[TimeSeries]] = None,
+    ):
+        """Fits the scorer on the given time series input.
+
+        If 2 time series are given, a reduced function, given as a parameter in the __init__ method
+        (reduced_function), will be applied to transform the 2 time series into 1. Default: "abs_diff"
+
+        Parameters
+        ----------
+        series_1
+            1st time series
+        series_2:
+            Optionally, 2nd time series
+
+        Returns
+        -------
+        self
+            Fitted Scorer.
+        """
+        list_series_1, list_series_2 = _convert_to_list(series_1, series_2)
+
+        if series_2 is None:
+            for idx, series in enumerate(list_series_1):
+                _check_if_TimeSeries(series)
+                list_series_1[idx], _ = self._check_probabilistic_case(series)
+        else:
+            for idx, (s1, s2) in enumerate(zip(list_series_1, list_series_2)):
+                _sanity_check_2series(s1, s2)
+                list_series_1[idx], _ = self._check_probabilistic_case(s1)
+                list_series_2[idx], _ = self._check_probabilistic_case(s2)
+
+        self._fit_core(list_series_1, list_series_2)
+
+    def _diff_sequence(
+        self, list_series_1: Sequence[TimeSeries], list_series_2: Sequence[TimeSeries]
+    ) -> Sequence[TimeSeries]:
+        """Calls the function _diff() on every pair (s1,s2) in the list (list_series_1,list_series_2).
+
+        list_series_1 and list_series_2 must have the same length n.
+        Each pair of series in list_series_1 and list_series_2 must be of the same length L and width/dimension W.
+
+        Parameters
+        ----------
+        series_1
+            1st sequence of time series
+        series_2:
+            2nd sequence of time series
+
+        Returns
+        -------
+        TimeSeries
+            Sequence of series of length n
+        """
+        list_series = []
```

Comment:
You seem to be working a lot with lists everywhere.
You could perhaps consider replacing those by generators in some places for efficiency. For instance here:
```python
return map(self._diff, zip(list_series_1, list_series_2))
```

This is more elegant than always relying on for loops, and can be more efficient in some cases as results are not proactively computed. You can instantiate a list at the very end (when returning the result to the user), which triggers the actual computation on the whole generator.